site/src/pages/release-notes/1.21.0.mdx

```diff
@@ -0,0 +1,150 @@
+    @Get("/fast/api")
+    public CompletableFuture<MyResponse> fastApi() {
+      ...
+    }
+
+    // A request gets timed out after 5 seconds.
+    @RequestTimeout(value = 5, unit = TimeUnit.SECONDS) // 👈👈👈
+    public CompletableFuture<MyResponse> slowApi() {
+      ...
+    }
+  }
+  ```
+- You can now set the default `MeterRegistry` by implementing a custom <type://FlagsProvider>. #4541
+  ```java
+  public class CustomFlagsProvider implements FlagsProvider {
+    @Override
+    public MeterRegistry meterRegistry() {
+      return PrometheusMeterRegistries.newRegistry(); // 👈👈👈
+    }
+  }
+  ```
+- You can now set the default <type://HttpHeaders> to <type://Server>, <type://VirtualHost> and
+  <type://Service>. #4520
+  ```java
+  static AsciiString CUSTOM_HEADER =
+    HttpHeaderNames.of("x-custom-header");
+
+  Server
+    .builder()
+    .setHeader(CUSTOM_HEADER, "server-default-header")
+    .virtualHost("foo.com")
+    .setHeader(CUSTOM_HEADER, "vhost-default-header").and()
+    .route()
+    .path("/foo")
+    .setHeader(CUSTOM_HEADER, "service-default-header")
+  ```
+- You can now set `:authority` or `Host` header whose value is different from <type://Endpoint#host()> using
+  <type://ClientBuilder#setHeader(CharSequence,Object)>. #4440 #4520
+  - This will be useful when you want to send a request through a proxy server.
+  ```java 
+  WebClient
+    // A request is sent to 'reverse-proxy.example.com'.
+    .builder("http://reverse-proxy.example.com/")
+    // but 'my-order.service.com' is used for the ':authority' header.
+    .setHeader(HeaderNames.AUTHORITY, "my-order.service.com")
+    ...
+  ```
+- <type://CircuitBreakerClient> can now be used with arbitrary <type://CircuitBreaker> implementations. #4426
+- Added <type://Bytes> that represents binary data. #4505
+- You can now see the description of services in the overview section of <type://DocService>. #4480 #4537
+- You can now override an authority of a gRPC client using `CallOptions`. #4445 #4520
+
+## 📈 Improvements
+
+- <type://DecodingClient> and <type://DecodingService> can now limit the maximum length of decompressed
+  data. #4469 #4564
+- `synchronized` blocks have been replaced with <type://ReentrantLock> for better virtual thread 
+  compatibility. #4551 #4549
+- <type://ServiceSpecification> for <type://DocService> are lazily loaded so as not to block a <type://Server>
+  from starting. #4491
+
+## 🛠️ Bug fixes
+
+- <type://HttpHeaders> set by <type://ClientBuilder#setHeader(CharSequence,Object)>,
+  <type://ClientBuilder#setHeader(CharSequence,Object)> and their variants are now stored in
+  <type://ClientRequestContext#defaultRequestHeaders()> which have the lowest priority and don't override the
+  values in <type://RequestHeaders>.
+  - Previously, the <type://HttpHeaders> set via <type://ClientBuilder> were stored in
+    <type://ClientRequestContext#additionalRequestHeaders()> which may override the values in
```

Comment:
past-tense since we are talking about the previous implementation

```suggestion
    <type://ClientRequestContext#additionalRequestHeaders()> which overrode the values in
```